can/bus.py

```diff
@@ -44,12 +44,19 @@ class BusABC(metaclass=ABCMeta):
     #: Log level for received messages
     RECV_LOGGING_LEVEL = 9
 
+    # Even if __init__ did not complete due to an exception, these attributes
+    # exist and won't cause weired problems in shutdown()
+    _is_initialized: bool = False
+    _is_shutdown: bool = False
```

Comment:
I'm not sure we need 2 separate attributes to implement this. If we only want `__del__` to execute if the class is initialized, perhaps `_is_active` would be more idiomatic?